tests/containers/volumes.pm

```diff
@@ -0,0 +1,110 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: podman network
+# Summary: Test podman network
+# Maintainer: qac team <qa-c@suse.de>
+
+use Mojo::Base 'containers::basetest';
+use testapi;
+use serial_terminal 'select_serial_terminal';
+
+sub test_volumes {
+    my %args = @_;
+    my $runtime = $args{runtime};
+
+    # From https://docs.docker.com/storage/bind-mounts/
+    # The --mount flag does not support z or Z options for modifying selinux labels.
+    my $Z = $runtime eq "podman" ? ",Z" : "";
+
+    my $test_dir = "test_dir";
+    my $test_file = "test_file";
+    my $test_image = "test_image";
+    my $test_container = "test_container";
+    my $test_volume = "test_volume";
+
+    # Create a directory with the sticky-bit set because of rootless podman
+    assert_script_run("mkdir -m 1777 $test_dir");
+
+    # Create Dockerfile with VOLUME defined
+    assert_script_run("echo -e 'FROM busybox\nVOLUME /$test_dir' > $test_dir/Dockerfile");
+
+    # Build image
+    assert_script_run("$runtime build -t $test_image -f $test_dir/Dockerfile $test_dir/");
+
+    assert_script_run("touch $test_dir/$test_file");
+
+    # Test --volumes-from
+    # Case 1: Check that the volume from container is visible in another container
+    assert_script_run("$runtime run -d --name $test_container $test_image");
+    assert_script_run("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir/$test_file");
+    assert_script_run("$runtime rm -vf $test_container");
+
+    # Case 2: Check that the volume from container is visible in another container, but the
+    # first container is mounting it in the directory specified as VOLUME in the Dockerfile
+    assert_script_run("$runtime run -d --name $test_container -v \$PWD/$test_dir:/$test_dir:Z $test_image");
+    # NOTE: When https://github.com/containers/podman/issues/19529 is solved we must add a version check here
+    my $ret = script_run("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir/$test_file");
+    record_soft_failure("gh#19529 - Unexpected error with --volumes-from") if ($ret != 0 && $runtime == "podman");
+    assert_script_run("$runtime rm -vf $test_container");
+
+    # Test --volume option with directory (read-only)
+    assert_script_run("! $runtime run --rm --volume \$PWD/$test_dir:/$test_dir:ro,Z $test_image rm /$test_dir/$test_file");
```

Comment:
```suggestion
    assert_script_run("! $runtime run --rm --volume \$PWD/$test_dir:/$test_dir:ro$Z $test_image rm /$test_dir/$test_file");
```
Didn't you mean `$Z` here?

Reply:
> Didn't you mean `$Z` here?

No because the `:Z` option is supported by both Podman and Docker with the `--volume` option.  It's only with the `--mount`  that Docker doesn't support.